# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsParent.cpp
# Commit: 74b39df74a4a
# Full Hash: 74b39df74a4a38e44f4b3783a5c4e4e8980ccf6f
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-06-18 16:17:05
# Description:
#   Bug 1903169 - Drop directory lock only when GetDatabasesOp owns one; r=dom-storage-reviewers,aiunusov
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D214070
# ==============================================================================

diff -r e2f0582a23f9 -r 74b39df74a4a dom/indexedDB/ActorsParent.cpp
--- a/dom/indexedDB/ActorsParent.cpp	Tue Jun 18 15:46:50 2024 +0300
+++ b/dom/indexedDB/ActorsParent.cpp	Tue Jun 18 13:01:45 2024 +0000
@@ -16737,7 +16737,7 @@
 
   mResolver(mDatabaseMetadataArray);
 
-  DropDirectoryLock(mDirectoryLock);
+  SafeDropDirectoryLock(mDirectoryLock);
 
   CleanupMetadata();
 
